modules/grpc/otel/otel-source-services.hpp

```diff
@@ -123,6 +123,7 @@ syslogng::grpc::otel::TraceServiceCall::Proceed(bool ok)
                 }
 
               LogMessage *msg = log_msg_new_empty();
+              msg_set_context(msg);
```

Comment:
We want to track `the message we are currently working on`. This idea will always imply a certain amount of layering violation, but placing this call into LogMessage constructors wouldn't necessarily help as I'm not sure we want to track each message that pops into existence.

For example, what do we want to do with synthetic messages produced by the `grouping-by()` and PatternDB parsers?

Reply:
Once this question is answered, I think we can introduce a function that can help reduce these set_context calls at least on the source side.